### ISSUE TITLE
Strange artifacts in the background on a rotated monitor

### ISSUE THREAD

**Issue by pastalian** [hyprwm/Hyprland#4453]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch HEAD at commit 6b92144f15117f9f7d93fe9867d6ec849d9abce1  (surface: avoid spam of window surfaces with scale and transform events).
Date: Thu Jan 11 22:07:28 2024
Tag: v0.34.0-40-g6b92144f

flags: (if any)


System Information:
System name: Linux
Node name: OwO
Release: 6.7.0-gentoo
Version: #1 SMP Tue Jan  9 00:53:02 JST 2024


GPU information: 
2d:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M/6850M XT] [1002:73df] (rev c5) (prog-if 00 [VGA controller])


os-release: NAME=Gentoo
ID=gentoo
PRETTY_NAME="Gentoo Linux"
ANSI_COLOR="1;32"
HOME_URL="https://www.gentoo.org/"
SUPPORT_URL="https://www.gentoo.org/support/"
BUG_REPORT_URL="https://bugs.gentoo.org/"
VERSION_ID="2.14"


plugins:
```

</details>

### Bug or Regression?

Bug

### Description

6b92144 causes strange artifacts in the background on a rotated monitor.

### How to reproduce

Rotate a monitor using `transform,X` .

monitor=DP-1, 2560x1440@143.912, 0x0, 1

### Crash reports, logs, images, videos

![24-01-16_133633](https://github.com/hyprwm/Hyprland/assets/28638872/80c75dc5-9d12-442a-a881-abd7b40ac9da)

**Comment by pastalian** [comment]
Fixed by 3e93fdf

**Comment by vaxerski** [comment]
my bad, should've mentioned.
